src/components/TextInput/TextInput.spec.tsx

```diff
@@ -140,6 +140,30 @@ describe("Text Input component", () => {
         cy.get(TEXT_INPUT_ID).should("have.attr", "autoComplete", "off");
     });
 
+    it("has the readonly prop turned off by default", () => {
+        mount(<StatefulInput />);
+        cy.get(TEXT_INPUT_ID).should("not.have.attr", "readonly");
+    });
+
+    it("turns the input readonly", () => {
+        mount(<StatefulInput readonly />);
+        cy.get(TEXT_INPUT_ID).should("have.attr", "readonly", "readonly");
+    });
+
+    it("still allows buttons to be pressed if readonly", () => {
+        mount(<StatefulInput readonly obfuscated copyable type={TextInputType.Password} value={PASSWORD} />);
+        cy.get(TEXT_INPUT_ID).should("have.attr", "readonly", "readonly");
```

Comment:
You don't have to test this twice, I'd either remove it here or keep it and remove the test above

Reply:
fixed :)